BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1703,33 +1709,37 @@ public void BeehiveDetonation()
-                                pBullet.HEType = PooledBulletTypes.Explosive;
-                                break;
-                            //legacy support for older configs that are still explosive = true
-                            case "true":
-                                pBullet.HEType = PooledBulletTypes.Explosive;
-                                break;
-                            case "shaped":
-                                pBullet.HEType = PooledBulletTypes.Shaped;
-                                break;
+                            string HEtype = sBullet.explosive.ToLower();
+                            switch (HEtype)
+                            {
+                                case "standard":
+                                    pBullet.HEType = PooledBulletTypes.Explosive;
+                                    break;
+                                //legacy support for older configs that are still explosive = true
+                                case "true":
+                                    pBullet.HEType = PooledBulletTypes.Explosive;
+                                    break;
+                                case "shaped":
+                                    pBullet.HEType = PooledBulletTypes.Shaped;
+                                    break;
+                            }
                         }
-                        pBullet.defaultDetonationRange = subDetonationRange;
-                        pBullet.detonationRange = subDetonationRange; // FIXME Use the default for now, maybe add a customisable value later?
+                        pBullet.detonationRange = subDetonationRange;
+                        //pBullet.defaultDetonationRange = defaultDetonationRange; //Deprecated legacy value functions have been spun off to TimeToDetonation/TimeToLive
                         pBullet.fuzeType = sFuze;
                         pBullet.timeToDetonation = sFuze switch
                         {
+                            BulletFuzeTypes.Timed => detonationRange / pBullet.bulletVelocity, //because beehive TimedFuze for the parent shell is timeToDetonation - detonationRange / bulletVelocity
                             BulletFuzeTypes.Flak => detonationRange / pBullet.bulletVelocity + Time.fixedDeltaTime, // Detonate at expected impact time for flak (plus 1 frame to allow proximity detection).
                             _ => pBullet.timeToLiveUntil - Time.time // Otherwise detonate at the TTL.
                         };
                     }
                     else
                     {
                         pBullet.fuzeType = BulletFuzeTypes.None;
-                        pBullet.sabot = ((sBullet.bulletMass * 1000 / (sBullet.caliber * sBullet.caliber * Mathf.PI / 400 * 19) + 1) * 10) > sBullet.caliber * 4;
+                        pBullet.sabot = isSabot;
                         pBullet.HEType = PooledBulletTypes.Slug;
                     }
                     pBullet.EMP = sBullet.EMP;
```

Comment:
Were there any other things that needed to get nulled/reset between submunition spawn and the bullet getting disabled/enabled next time its drawn from the Pool?

Reply:
Other than the `beehive` option that I set to `false`, none that I'm aware of, but it's always possible that I missed some. I only noticed the `beehive` one due them appearing from the explosion of a submunition.